### DISCUSSION TITLE
ROHD `withset` multiple bits at once

### DISCUSSION THREAD

**Discussion by quekyj** [intel/rohd#309] [Q&A] [answered]
Is it possible to access a range of indices in ROHD using syntax similar to `Logic a[2:4] = '01'` like Python? The ROHD example,
`e = e.withSet(3, d)` function only changes a single bit. What if we want to change multiple bits at once? Is it possible to have syntax similar to python like above?

**Comment by chykon** [comment, marked as answer]
Tried in [DartPad](https://dartpad.dev/?id=76230ee923024be6a8f1337effc3f592) - worked fine.

**Reply by mkorbel1** [reply]
Good example, thanks!  Yes, `withSet` will infer the number of bits to apply based on the width of the signal being applied.

**Comment by quekyj** [comment]
Looks like I made some mistakes about that. Let me edit the question.

**Reply by chykon** [reply]
There are some restrictions related to the fact that this syntax is not planned to be added to Dart: <https://github.com/dart-lang/sdk/issues/983#issuecomment-842176477>.

**Reply by quekyj** [reply]
I see. thanks!
